pkg/util/winutil/shutil.go

```diff
@@ -106,3 +109,39 @@ func GetProgramDataDirForProduct(product string) (path string, err error) {
 	path = val
 	return
 }
+
+// GetProgramFilesDirForProduct returns the root of the installatoin directory,
+// usually c:\program files\datadog\datadog agent
+func GetProgramFilesDirForProduct(product string) (path string, err error) {
+	keyname := "SOFTWARE\\Datadog\\" + product
+	k, err := registry.OpenKey(registry.LOCAL_MACHINE,
+		keyname,
+		registry.ALL_ACCESS)
+	if err != nil {
+		// if the key isn't there, we might be running a standalone binary that wasn't installed through MSI
+		log.Debugf("Windows installation key root (%s) not found, using default program data dir", keyname)
+		return getDefaultProgramFilesDir()
+	}
+	defer k.Close()
+	val, _, err := k.GetStringValue("InstallPath")
+	if err != nil {
+		log.Warnf("Windows installation key config not found, using default program data dir")
+		return getDefaultProgramFilesDir()
+	}
+	path = val
+	return
+}
+
+func getDefaultProgramFilesDir() (path string, err error) {
+	var retstr uintptr
+	err = SHGetKnownFolderPath(&FOLDERIDProgramFiles, 0, 0, &retstr)
```

Comment:
I think you can use `windows.KnownFolderPath(windows.FOLDERID_ProgramFiles, 0)` instead. This would allow you to get rid of the redefinition of `FOLDERIDProgramFiles`. `getDefaultProgramDataDir` could also be updated.

Reply:
true.  Those weren't there when I originally implemented this.  It does expand the scope of the PR.